src/services/sync/strategies/WebUntis/WebUntisSchoolyearSyncer.js

```diff
@@ -0,0 +1,483 @@
+		// And reflect changes in metadata store
+		const webUntisMetadataService = this.app.service('webuntisMetadata');
+
+		const importCondition = this.getImportConditionEvaluator(params);
+		const promises = data.map((entry) => {
+			if (importCondition(entry.state.toString())) {
+				return this.obtainAndUpdateCourseAndClass(entry)
+					.then(() => webUntisMetadataService.patch(entry._id, { state: 'imported' }))
+					.catch(() => webUntisMetadataService.patch(entry._id, { state: 'errored' }));
+			} else {
+				return webUntisMetadataService.patch(entry._id, { state: 'discarded' });
+			}
+		});
+
+		await Promise.all(promises);
+	}
+
+	/**
+	* Entry: {
+	* 	 datasourceId: this.data.datasourceId,
+	* 	 teacher: 'Renz',
+	* 	 class: '2a',
+	* 	 room: '0-23',
+	* 	 subject: 'mathe',
+	* 	 state: 'new',
+	* }
+	*/
+	async obtainAndUpdateCourseAndClass(entry) {
+		const courseService = this.app.service('courses');
+		const classService = this.app.service('classes');
+
+		const school = await this.getSchool(await this.getUser());
+
+		/**
+        * Mapping:
+        *
+        * Schul-Cloud: class, course (per class), lesson
+        * WebUntis: class, subject
+        * German: Klasse, Kurs, Fach, Schulstunde
+        */
+
+		// Get class
+		const scClasses = await classService.find({ query: { name: entry.class }, paginate: false });
+		
+		let scClass = scClasses[0];
+		if (scClass === undefined) {
+			// Create Schul-Cloud class?
+			const newClass = {
+				name: className,
+				schoolId: school._id,
+				nameFormat: 'static',
+				year: school.currentYear,
+			};
+			scClass = await classService.create(newClass);
+		}
+
+		// Get course
+		const scCourses = await courseService.find({
+			query: {
+				name: entry.subject,
+				classIds: scClass._id,
+				schoolId: school._id,
+			},
+			paginate: false,
+		});
+
+		scCourse = scCourses[0];
+		if (scCourse === undefined) {
+			// Create Course
+			const newCourse = {
+				name: courseName,
+				classIds: [scClass._id],
+				schoolId: school._id,
+				teacherIds: [],
+			};
+
+			scCourse = await courseService.create(newCourse);
+		}
+
+		// Get times
+		
+		const times = [];
+		for (const timetableEntry of entry.timetable) {
+			const newEntry = {
+				weekday: this.getWeekDay(timetableEntry.date),
+				startTime: this.getStartTime(timetableEntry.startTime),
+				duration: this.getDuration(timetableEntry.startTime, timetableEntry.endTime),
+				room: timetableEntry.room,
+				count: 1,
+			};
+			let entryFound = false;
+			for (const givenEntry of times) {
+				if (givenEntry.weekday === newEntry.weekday
+					&& givenEntry.startTime === newEntry.startTime
+					&& givenEntry.duration === newEntry.duration
+					&& givenEntry.room === newEntry.room) {
+					givenEntry.count += 1;
+					entryFound = true;
+				}
+			}
+			if (!entryFound) {
+				times.push(newEntry);
+			}
+		}
+
+		// Assumption: After 2 entries it is recurring
+		times = times.filter(entry => entry.count >= 2).map(entry => ({
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['entry' is already declared in the upper scope.](https://app.codacy.com/manual/schul-cloud/schulcloud-server/pullRequest?prid=4061937)